### PR TITLE
test(collaboration): add e2e test for remote collaboration and fix bugs

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/cli/tests/e2e/collaboration/Permission.tests.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    const listCollaboratorResult = await execAsync(
+      `teamsfx permission status --list-all-collaborator`,
+      {
+        cwd: projectPath,
+        env: process.env,
+        timeout: 0,
+      }
+    );
+
+    expect(listCollaboratorResult.stdout).to.contains(`Account: ${collaborator?.split("@")[0]}`);
```
Comment (SLdragon) [inline packages/cli/tests/e2e/collaboration/Permission.tests.ts:88]: Add some comments about `collaborator?.split("@")[0]`, and also check collaborator list also contains creator email

**Review by Alive-Fish — APPROVED** [review]
lgtm

File: packages/cli/tests/e2e/collaboration/Permission.tests.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    // e.g. Guest account "account@example.com" will appear as "account_example#EXT#@exampleTenant.onmicrosoft.com" under tenant "exampleTenant".
+    // Thus here will check the account name only.
+    expect(listCollaboratorResult.stdout).to.contains(`Account: ${collaborator?.split("@")[0]}`);
+    expect(listCollaboratorResult.stdout).to.contains(`Account: ${creator?.split("@")[0]}`);
+    console.log("[Successfully] list collaborator");
+  });
+
+  after(async () => {
+    // clean up
+    // await cleanUp(appName, projectPath, true, false, false);
```
Comment (SLdragon) [inline packages/cli/tests/e2e/collaboration/Permission.tests.ts:100]: This should uncomment

Comment (KennethBWSong) [reply packages/cli/tests/e2e/collaboration/Permission.tests.ts]: Fixed. Thanks!

**Review by SLdragon — APPROVED** [review]

**Review by LongOddCode — APPROVED** [review]

**Pull request merged**